Airship config bundler broken since 0.11

The config bundler is broken after 0.11, but worked in 0.10.

There are two issues; first you can not initialize a new config repo:

```
$ asconfig-0.12 init --groupId com.example.config --releasesRepository nexus --snapshotsRepository nexus --releasesRepositoryUri 'http://192.168.193.6:8082/nexus/content/repositories/releases' --snapshotsRepositoryUri 'http://192.168.193.6:8082/nexus/content/repositories/snapshots'
Exception in thread "main" java.lang.IllegalArgumentException: io.airlift.airship.configbundler.Metadata could not be converted to json
    at io.airlift.json.JsonCodec.toJson(JsonCodec.java:142)
    at io.airlift.airship.configbundler.Metadata.save(Metadata.java:48)
    at io.airlift.airship.configbundler.Model.initialize(Model.java:88)
    at io.airlift.airship.configbundler.InitCommand.call(InitCommand.java:57)
    at io.airlift.airship.configbundler.InitCommand.call(InitCommand.java:12)
    at io.airlift.airship.configbundler.ConfigBundler.main(ConfigBundler.java:30)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Class com.fasterxml.jackson.databind.ser.BeanPropertyWriter can not access a member of class io.airlift.airship.configbundler.Metadata with modifiers "public" (through reference chain: io.airlift.airship.configbundler.Metadata["groupId"])
    at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:232)
    at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:197)
    at com.fasterxml.jackson.databind.ser.std.StdSerializer.wrapAndThrow(StdSerializer.java:183)
    at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:559)
    at com.fasterxml.jackson.databind.ser.BeanSerializer.serialize(BeanSerializer.java:143)
    at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:120)
    at com.fasterxml.jackson.databind.ObjectMapper._configAndWriteValue(ObjectMapper.java:2623)
    at com.fasterxml.jackson.databind.ObjectMapper.writeValueAsString(ObjectMapper.java:2119)
    at io.airlift.json.JsonCodec.toJson(JsonCodec.java:139)
    ... 5 more
Caused by: java.lang.IllegalAccessException: Class com.fasterxml.jackson.databind.ser.BeanPropertyWriter can not access a member of class io.airlift.airship.configbundler.Metadata with modifiers "public"
    at sun.reflect.Reflection.ensureMemberAccess(Reflection.java:95)
    at java.lang.reflect.AccessibleObject.slowCheckMemberAccess(AccessibleObject.java:261)
    at java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:253)
    at java.lang.reflect.Method.invoke(Method.java:594)
    at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.get(BeanPropertyWriter.java:653)
    at com.fasterxml.jackson.databind.ser.BeanPropertyWriter.serializeAsField(BeanPropertyWriter.java:509)
    at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFields(BeanSerializerBase.java:551)
    ... 10 more
```

Second, you can not work with a currently created config repo:

```
$ asconfig-0.12 snapshot
Exception in thread "main" java.lang.IllegalArgumentException: Invalid [simple type, class io.airlift.airship.configbundler.Metadata] json string
    at io.airlift.json.JsonCodec.fromJson(JsonCodec.java:124)
    at io.airlift.airship.configbundler.Model.readMetadata(Model.java:65)
    at io.airlift.airship.configbundler.SnapshotCommand.call(SnapshotCommand.java:30)
    at io.airlift.airship.configbundler.SnapshotCommand.call(SnapshotCommand.java:16)
    at io.airlift.airship.configbundler.ConfigBundler.main(ConfigBundler.java:30)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Could not find creator property with name 'groupId' (in class io.airlift.airship.configbundler.Metadata)
 at [Source: java.io.StringReader@19fc0a04; line: 1, column: 1]
    at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:164)
    at com.fasterxml.jackson.databind.DeserializationContext.mappingException(DeserializationContext.java:608)
    at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.addBeanProps(BeanDeserializerFactory.java:551)
    at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.buildBeanDeserializer(BeanDeserializerFactory.java:267)
    at com.fasterxml.jackson.databind.deser.BeanDeserializerFactory.createBeanDeserializer(BeanDeserializerFactory.java:171)
    at com.fasterxml.jackson.databind.deser.DeserializerCache._createDeserializer(DeserializerCache.java:388)
    at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCache2(DeserializerCache.java:264)
    at com.fasterxml.jackson.databind.deser.DeserializerCache._createAndCacheValueDeserializer(DeserializerCache.java:244)
    at com.fasterxml.jackson.databind.deser.DeserializerCache.findValueDeserializer(DeserializerCache.java:143)
    at com.fasterxml.jackson.databind.DeserializationContext.findRootValueDeserializer(DeserializationContext.java:342)
    at com.fasterxml.jackson.databind.ObjectMapper._findRootDeserializer(ObjectMapper.java:2895)
    at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:2789)
    at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:1957)
    at io.airlift.json.JsonCodec.fromJson(JsonCodec.java:121)
    ... 4 more
'''
```

--- a/airship-config-bundler/src/main/java/io/airlift/airship/configbundler/Metadata.java
+++ b/airship-config-bundler/src/main/java/io/airlift/airship/configbundler/Metadata.java
@@ -1,5 +1,6 @@
 package io.airlift.airship.configbundler;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.base.Charsets;
 import com.google.common.base.Preconditions;
@@ -9,12 +10,13 @@ import io.airlift.json.JsonCodec;
 import java.io.File;
 import java.io.IOException;
 
-class Metadata
+public class Metadata
 {
     private final String groupId;
     private final Repository snapshotsRepository;
     private final Repository releasesRepository;
 
+    @JsonCreator
     public Metadata(@JsonProperty("groupId") String groupId,
             @JsonProperty("snapshotsRepository") Repository snapshotsRepository,
             @JsonProperty("releasesRepository") Repository releasesRepository)
@@ -54,6 +56,7 @@ class Metadata
         private final String id;
         private final String uri;
 
+        @JsonCreator
         public Repository(@JsonProperty("id") String id, @JsonProperty("uri") String uri)
         {
             Preconditions.checkNotNull(id, "id is null");

